File: task/src/main/java/org/kharitonov/springWeb/dto/factories/TaskDTOFactory.java
package org.kharitonov.springWeb.dto.factories;

import org.kharitonov.springWeb.dto.TaskDTO;
import org.kharitonov.springWeb.entites.TaskEntity;
import org.springframework.stereotype.Component;

@Component
public class TaskDTOFactory {

    public TaskDTO toTaskDTO(TaskEntity taskEntity) {
        return new TaskDTO.Builder()
                .id(taskEntity.getId())
                .name(taskEntity.getName())
                .createAt(taskEntity.getCreateAt())
                .description(taskEntity.getDescription())
                .build();
    }
}

File: laba_chetyre/src/main/kotlin/by/bsut/labachetyre/dto/factories/TaskDTOFactory.kt
package by.bsut.labachetyre.dto.factories

import by.bsut.labachetyre.dto.TaskDTO
import by.bsut.labachetyre.entites.TaskEntity
import org.springframework.stereotype.Component

@java.awt.Component
class TaskDTOFactory {

    fun toTaskDTO(taskEntity: TaskEntity): TaskDTO = Builder().id(taskEntity.getId())
        .name(taskEntity.getName())
        .createAt(taskEntity.getCreateAt())
        .description(taskEntity.getDescription())
        .build()
}